lib/knock.rb

```diff
@@ -16,8 +16,18 @@ module Knock
   self.token_signature_algorithm = "HS256"
 
   # Configure the key used to sign tokens.
-  mattr_accessor :token_secret_signature_key
-  self.token_secret_signature_key = -> { Rails.application.secrets.secret_key_base }
+  mattr_writer :token_secret_signature_key, default: -> { Rails.application.secrets.secret_key_base }
+
+  class EmptySecretKey < StandardError
+    def initialize(msg="Knock secret signature key can't be empty")
+      super
+    end
+  end
+
+  def self.token_secret_signature_key
```

Comment:
I think it would be better raising the error in the assign moment (setter) instead of the reading moment (getter). It's better to prevent setter from accepting a wrong value than accepting it and causing errors when you try to use that value... So, I'd suggest to invert it, creating a method for the setter, and using `mattr_reader` for the getter...